lib/utils/utilities.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright (c) 2024, STMicroelectronics
+ *
+ */
+
+#include <internal/utilities.h>
+#include <metal/io.h>
+#include <metal/utilities.h>
+
+size_t safe_strcpy(char *dst, size_t d_size, const char *src, size_t s_size)
+{
+	size_t size = metal_min(s_size, d_size);
+	size_t nleft = size + 1;
+	char *d = dst;
+
+	/* Copy as many bytes as will fit. */
+	if (nleft != 0) {
```

Comment:
Can this test ever be false? size_t can never be negative, so based on the assignment above, nleft must always be at least 1.